change(shell): use libsql-shell-go library was shell

luisfvieirasilva opened the pull request:
`turso db shell` using libsql-shell-go lib.

Notes:
- `libsql-shell-go` is a private repo for now. So to test we need to configure it correctly (see instructions below)

----

How to add libsql-shell-go private module:
- Tell Go that libsql-shell-go is a private module, so it shouldn't search for it at https://sum.golang.org/
  - `export GOPRIVATE="github.com/libsql/libsql-shell-go"`
- Configure your GitHub to always use SSH instead of HTTPS for GitHub. You need to configure a SSH Key and add them to an agent (the key password must be saved by the agent before importing the module)
  - [Link](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) about how configure SSH Key
  - At `~/.gitconfig` (doesn't work if configure repo's .git/config), add following lines
```
[url "ssh://git@github.com/"]
    insteadOf = https://github.com/
```
- Get mod packages
  - `go mod download`

athoscouto reviewed (changes requested):
`turso-cli` is public, it doesn't make much sense to add a private dependency to it.
Please open source the dependency before making this change.

glommer commented:
I am testing this and will add comments as I find them.
First one:
* this lost the spinner that @athoscouto put so much effort into cooking and was a nice DX, so we should add it back

haaawk commented on internal/cmd/db_show.go:
```diff
@@ -49,11 +49,6 @@ var showCmd = &cobra.Command{
 			return nil
 		}
 
-		if showHttpUrlFlag {
```
Why is this removed?

haaawk commented on internal/cmd/db_shell.go:
```diff
@@ -188,224 +125,19 @@ func tokenFromDb(db *turso.Database, client *turso.Client) (string, error) {
-	}
-	return http.DefaultClient.Do(req)
+func addTokenAsQueryParameter(dbUrl string, token string) string {
+	return fmt.Sprintf("%s/?jwt=%s", dbUrl, token)
```
Do we need `/?`? I thought `/` is just needed in zsh to escape `?` but here it shouldn't be needed, no?

luisfvieirasilva replied:
It's not zsh escape, it's just a optional final slash in the end of a path that I'm used to add, but it's unnecessary

haaawk commented on internal/cmd/db_shell.go:
```diff
@@ -67,117 +46,75 @@ var shellCmd = &cobra.Command{
 		}
 
 		dbUrl := nameOrUrl
-		if db != nil {
```
Why is it now safe not to check whether db is not nil?

luisfvieirasilva replied:
I forced an error inside `databaseFromNameOrURL` if we can't find the database

haaawk commented on internal/cmd/db_shell.go:
```diff
@@ -46,12 +31,6 @@ var shellCmd = &cobra.Command{
 		}
 		cmd.SilenceUsage = true
 
-		spinner := prompt.StoppedSpinner("Connecting to database")
```
This needs to be brought back, right?

luisfvieirasilva commented:
> What's blocking this @luisfvieirasilva ? Have you addressed all the comments?

I was changing libsql-client-go to be able to use `http` connections here. I'll go back to this once the [PR](https://github.com/libsql/libsql-client-go/pull/26) is merged

luisfvieirasilva reviewed (commented):
@haaawk @athoscouto @glommer PR updated

luisfvieirasilva commented on internal/cmd/db_shell.go:
```diff
@@ -76,26 +63,34 @@ var shellCmd = &cobra.Command{
+		connectionInfo := getConnectionInfo(nameOrUrl, db, config)
+
+		shellConfig := shell.ShellConfig{
+			DbPath:         dbUrl,
```
@athoscouto, new two tickets based on this comment

- [Improve DbPath name](https://github.com/libsql/libsql-shell-go/issues/87)
- [Add option to pass Bearer token as ShellConfig](https://github.com/libsql/libsql-shell-go/issues/88)

luisfvieirasilva commented on internal/cmd/db_shell.go:
```diff
@@ -48,8 +33,10 @@ var shellCmd = &cobra.Command{
+		spinnerStarted := false
 		if len(args) == 1 {
 			spinner.Start()
+			spinnerStarted = true
```
done!
